examples/cis2-wccd/src/lib.rs

```diff
@@ -110,6 +110,17 @@ struct SetImplementorsParams {
     implementors: Vec<ContractAddress>,
 }
 
+/// The parameter type for the contract function `upgrade`.
+/// Takes the new module and optionally a migration function to call in the new
```

Comment:
I'm not sure if we should phrase it as 'migration function' maybe it will lead to confusion of what that is? Maybe just call it 'function' and then below document that this could e.g. be used for 'migration' behavior. 

IMO. The key take away from this documentation should be that one can  invoke a function with the new 'upgraded' context within the same transaction as the upgrade itself.